Implementation section and build update (requires manual merge)

gadams999 opened the pull request:
Description of changes:
- Content addition for new Implementation section
- Support for rendering PlantUML in developer mode
- Dockerfile multi-stage to reduce local image size
- Removed unused themes
- Updated `children` shortcode for `page` filtering support

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

TheRyanBurke commented:
1. The interim page at /en/implementations/aws/command/ has no content. At a minimum, there should be a link to the singular child page. Let's establish a convention for these kinds of pages as there will be more to follow.
2. The sample topics used in the Command implementation do not reflect the architectural guidance of the whitepaper at https://d1.awsstatic.com/whitepapers/Designing_MQTT_Topics_for_AWS_IoT_Core.pdf For example, command topics would be prefixed with `cmd/` and look something like `cmd/<application>/<context>/<destination-id>/<req-type>`. Shall we conform to the guidance in this whitepaper for AWS implementations in the Atlas?
3. Typo in bullet 1 under Reference Architecture "and the subscribes to the"
4. Verb tense mismatch in bullet 2 under Reference Architecture "The Application established". Surrounding content uses present tense e.g. "The Device establishes".
5. "The code samples focus the command design" -> "The code samples focus on the command design" ?
6. Device section writes "and then publishing the outcome of each command back to the Application." Should we rephrase to indicate that the device publishes a response to the broker with the intent that it reaches the application?
7. The included Considerations have more to do with the pattern itself than the provided implementation. Should we consider migrating these to the respective Pattern page?
8. Recommend including a reference to the Designing MQTT Topics whitepaper somewhere in the section titled Implementation. Perhaps an addition to the first paragraph or a second Note tooltip.
9. The section at /en/contribute/#guidelines-for-content-creation uses the same header level as the sections before and after it, but does not include any content. Should this be an h1 instead of an h2?

I did not critically review the files under src/docker_assets/.

gadams999 commented:
> 1. The interim page at /en/implementations/aws/command/ has no content. At a minimum, there should be a link to the singular child page. Let's establish a convention for these kinds of pages as there will be more to follow.

Added list of _command_ only articles. Will replicate for others. Concept is to use use `implementations/aws/command` as the main directory, then each implementation a increasing directory under that such as `command/command1`, `command/command2`, etc. Keeping an implementations content (`_index.md` plus graphics) provides the ability to move.

> 2. The sample topics used in the Command implementation do not reflect the architectural guidance of the whitepaper at https://d1.awsstatic.com/whitepapers/Designing_MQTT_Topics_for_AWS_IoT_Core.pdf For example, command topics would be prefixed with cmd/ and look something like cmd/<application>/<context>/<destination-id>/<req-type>. Shall we conform to the guidance in this whitepaper for AWS implementations in the Atlas?

Corrected in graphic, article, and code samples

> 3. Typo in bullet 1 under Reference Architecture "and the subscribes to the"

Corrected. Both now read _and then subscribes_

> 4. Verb tense mismatch in bullet 2 under Reference Architecture "The Application established". Surrounding content uses present tense e.g. "The Device establishes".

Changed to same tense as previous bullet point.

> 5. "The code samples focus the command design" -> "The code samples focus on the command design" ?

Corrected.

> 6. Device section writes "and then publishing the outcome of each command back to the Application." Should we rephrase to indicate that the device publishes a response to the broker with the intent that it reaches the application?

Expanded description of the process.

> 7. The included Considerations have more to do with the pattern itself than the provided implementation. Should we consider migrating these to the respective Pattern page?

I think these are still specific to AWS IoT Core. Fore instance, duplicate messages still need to be addressed as AWS IoT Core only support QoS 0 and 1, but not 2 (exactly once messages). Agreed these could be more specific.

> 8. Recommend including a reference to the Designing MQTT Topics whitepaper somewhere in the section titled Implementation. Perhaps an addition to the first paragraph or a second Note tooltip.

Added after first paragraph.
> 9. The section at /en/contribute/#guidelines-for-content-creation uses the same header level as the sections before and after it, but does not include any content. Should this be an h1 instead of an h2?

WIP. Added paragraph for context.

TheRyanBurke commented:
Approved for merge
